packages/ruby/lib/readme/har/request_serializer.rb

```diff
@@ -25,6 +26,16 @@ def as_json
 
       private
 
+      def url
+        url = URI(@request.url)
+        headers = @request.headers
+        forward_proto = headers["X-Forwarded-Proto"]
+        forward_host = headers["X-Forwarded-Host"]
+        url.host = forward_host if forward_host.instance_of?(String)
+        url.scheme = forward_proto if forward_proto.instance_of?(String)
```

Comment:
Can `@request.headers` ever use subclassed strings? Would it be better to use `.kind_of?`

Reply:
I don't think so, but I'm happy to just in case